#!/usr/bin/env python
PACKAGE = "alex_mpc"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("rudder_granularity", double_t, 0, "Granularity at which to consider rudders", 0.0625, 1e-5, 1)
gen.add("throttle_granularity", double_t, 0, "Granularity at which to consider throttles", 0.125, 1e-5, 1)

gen.add("distance_weight", double_t, 0, "Weight of Euclidean distance (in meters) in reference trajectory comparison",
        1, 0, 10)
gen.add("heading_weight", double_t, 0, "Weight of heading difference (in radians) in reference trajectory comparison",
        20, 0, 60)
gen.add("speed_weight", double_t, 0, "Weight of speed difference (in m/s) in reference trajectory comparison", 5, 0, 10)

gen.add("achievable_threshold", double_t, 0, "Score threshold below which the reference trajectory is considered " +
        "achievable. Score is the weighted sum distance, heading difference, and speed difference", 20, 0, 1000)

gen.add("current_estimation", bool_t, 0, "Whether current estimation is enabled", True)

gen.add("timeout", double_t, 0, "Maximum duration after a reference trajectory is received for which controls are "
                                "still issued (seconds)", 5, 2, 3600)

exit(gen.generate(PACKAGE, "alex_mpc", "alex_mpc"))
